src/GenericContainer/include/BipedalLocomotionControllers/GenericContainer.h

```diff
@@ -0,0 +1,289 @@
+    {
+        m_span = span;
+        m_resizeLambda = resizeLambda;
+    }
+
+    GenericContainer(iDynTree::Span<T> span)
+    {
+        m_span = span;
+        m_resizeLambda = [span](index_type size){return span;};
+    }
+
+    ~GenericContainer() = default;
+
+    GenericContainer(const GenericContainer<T>& other) = delete; //It would not be clear if you are copying the pointers or the content.
+
+    GenericContainer(GenericContainer<T>&& other)
+    {
+        m_span = other.m_span;
+        m_resizeLambda = other.m_resizeLambda;
+    }
+
+    bool clone(const GenericContainer<T>& other)
+    {
+        if (size() != other.size())
+        {
+            if (!resize(other.size()))
+            {
+                std::cerr << "[Generic container] Failed to resize. Copy aborted" << std::endl;
+                return false;
+            }
+        }
+
+        for (index_type i = 0; i < size(); ++i)
+        {
+            this->operator[](i) = other[i];
+        }
+
+        return true;
+    }
+
+    void operator=(const GenericContainer<T>& other)
+    {
+        clone(other);
+    }
+
+    void operator=(GenericContainer<T>&& other)
+    {
+        clone(other);
+    }
+
+
+    bool resize(index_type newSize)
+    {
+        m_span = m_resizeLambda(newSize);
+        return m_span.size() == newSize;
+    }
+
+    index_type size() const
+    {
+        return m_span.size();
+    }
+
+    bool empty() const {
+        return m_span.empty();
+    }
+
+    reference operator[](index_type idx) const
+    {
+        return m_span[idx];
+    }
+
+    double getVal(index_type idx) const
+    {
+        return this->operator[](idx);
+    }
+
+    bool setVal(index_type idx, double val)
+    {
+        if (idx >= 0 && idx < size())
+        {
+            this->operator[](idx) = val;
+            return true;
+        }
+        return false;
+    }
+
+    reference at(index_type idx) const
+    {
+        return this->operator[](idx);
+    }
+
+    reference operator()(index_type idx) const
+    {
+        return this->operator[](idx);
+    }
+
+    pointer data() const
+    {
+        return m_span.data();
+    }
+
+    // [span.iter], span iterator support
+    iterator begin() const
+    {
+        return m_span.begin();
+    }
+    iterator end() const
+    {
+        return m_span.end();
+    }
+
+    const_iterator cbegin() const
+    {
+        return m_span.cbegin();
+    }
+
+    const_iterator cend() const
+    {
+        return m_span.cend();
+    }
+
+    reverse_iterator rbegin() const
+    {
+        return m_span.rbegin();
+    }
+
+    reverse_iterator rend() const
+    {
+        return m_span.rend();
+    }
+
+    const_reverse_iterator crbegin() const
+    {
+        return m_span.crbegin();
+    }
+
+    const_reverse_iterator crend() const
+    {
+        return m_span.crend();
+    }
+
+};
+
+namespace BipedalLocomotionControllers {
+
+    enum class GenericContainerMode
```

Comment:
According to the indentation style of the repo, the content of a `namespace` should not be indented 

You can load the `clang-format` file in your editor to be compliant with the style. For QTCreator you can use [this guide](https://www.vikingsoftware.com/using-clang-format-with-qtcreator/)

Reply:
That's me fighting against it :grin:. I was used to having an additional indentation level and I added it manually. I will remove it!

Reply:
I removed the indentation in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/29/commits/2fd26da1dcaac5814f255c8e7d2ba2cea0dcae15